tests/run_test.go

```diff
@@ -0,0 +1,210 @@
+        startup: enabled
+`,
+		filepath.Join(pebbleDir, "svc1"),
+	)
+	createLayer(t, pebbleDir, "001-simple-layer.yaml", layerYAML)
+
+	_, _ = pebbleRun(t, pebbleDir, "--hold")
+
+	// Sleep 100 millisecond before checking services because immediate check
+	// can't guarantee that svc1 is not started shortly after the log "Started daemon".
+	time.Sleep(100 * time.Millisecond)
+
+	_, err := os.Stat(filepath.Join(pebbleDir, "svc1"))
+	if err == nil {
+		t.Fatal("pebble run --hold failed, services are still started")
+	}
+	if !errors.Is(err, fs.ErrNotExist) {
+		t.Fatalf("Stat returned error other than ErrNotExist: %v", err)
+	}
+}
+
+func TestHTTPPort(t *testing.T) {
+	pebbleDir := t.TempDir()
+
+	port := "61382"
+	_, stderrCh := pebbleRun(t, pebbleDir, "--http=:"+port)
+	waitForLog(t, stderrCh, "Started daemon", 3*time.Second)
+
+	conn, err := net.Listen("tcp", ":"+port)
+	if err == nil {
+		conn.Close()
+		t.Fatalf("port %s is not being listened: %v", port, err)
+	}
+	if conn != nil {
+		conn.Close()
+	}
+
+	resp, err := http.Get(fmt.Sprintf("http://localhost:%s/v1/health", port))
+	if err != nil {
+		t.Fatalf("port %s is not being listened by : %v", port, err)
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != 200 {
+		t.Fatalf("error checking pebble /v1/health on port %s: %v", port, err)
+	}
+}
+
+func TestVerbose(t *testing.T) {
+	pebbleDir := t.TempDir()
+
+	layersFileName := "001-simple-layer.yaml"
+	layerYAML := `
+services:
+    svc1:
+        override: replace
+        command: /bin/sh -c "echo 'hello world'; sleep 1000"
+        startup: enabled
+`
+	createLayer(t, pebbleDir, layersFileName, layerYAML)
+
+	stdoutCh, stderrCh := pebbleRun(t, pebbleDir, "--verbose")
+	waitForLog(t, stderrCh, "Started daemon", 3*time.Second)
+	waitForLog(t, stdoutCh, "hello world", 3*time.Second)
+}
+
+func TestArgs(t *testing.T) {
+	pebbleDir := t.TempDir()
+
+	layerYAML := `
+services:
+    svc1:
+        override: replace
+        command: /bin/sh
+        startup: enabled
+`
+	layersFileName := "001-simple-layer.yaml"
+	createLayer(t, pebbleDir, layersFileName, layerYAML)
+
+	stdoutCh, stderrCh := pebbleRun(t, pebbleDir, "--verbose",
+		"--args",
+		"svc1",
+		"-c",
+		"echo 'hello world'; sleep 1000",
+	)
+	waitForLog(t, stderrCh, "Started daemon", 3*time.Second)
+	waitForLog(t, stdoutCh, "hello world", 3*time.Second)
+}
+
+func TestIdentities(t *testing.T) {
+	pebbleDir := t.TempDir()
+
+	identitiesYAML := `
+identities:
+    bob:
+        access: admin
+        local:
+            user-id: 42
+    alice:
+        access: read
+        local:
+            user-id: 2000
+`
+	identitiesFileName := "idents-add.yaml"
+	if err := os.WriteFile(filepath.Join(pebbleDir, identitiesFileName), []byte(identitiesYAML), 0o755); err != nil {
+		t.Fatalf("Cannot write identities file: %v", err)
+	}
+
+	_, _ = pebbleRun(t, pebbleDir, "--identities="+filepath.Join(pebbleDir, identitiesFileName))
+	time.Sleep(100 * time.Millisecond)
```

Comment:
I think we should wait for the "Starting daemon" log here, rather than a time.Sleep. The time.Sleep is okay in the "directories *not* created" case, but here we're waiting for something positive (not lack of something).